src/replica/replica.cpp

```diff
@@ -176,6 +176,19 @@ void replica::on_client_read(dsn::message_ex *request, bool ignore_throttling)
         response_client_read(request, ERR_ACL_DENY);
     }
 
+    // validate partition_hash
+    if (_validate_partition_hash) {
+        if (_split_mgr->should_reject_request()) {
+            response_client_read(request, ERR_SPLITTING);
+            return;
+        }
+        if (!_split_mgr->check_partition_hash(
+                ((dsn::message_ex *)request)->header->client.partition_hash, "read")) {
+            response_client_read(request, ERR_PARENT_PARTITION_MISUSED);
+            return;
+        }
+    }
```

Comment:
```cpp
bool replica::reject_read_if_splitting(message_ex* request) {
    if (_validate_partition_hash) {
        if (_split_mgr->should_reject_request()) {
            response_client_read(request, ERR_SPLITTING);
            return true;
        }
        if (!_split_mgr->check_partition_hash(
                ((dsn::message_ex *)request)->header->client.partition_hash, "read")) {
            response_client_read(request, ERR_PARENT_PARTITION_MISUSED);
            return true;
        }
    }
    return false;
}
```

Separating the new code in a func is more readable. And of course it doesn't have to be a macro. Why you think that?

```
if (reject_read_if_splitting()) {
  return;
}
```